pkg/management/component.go

```diff
@@ -1386,6 +1416,50 @@ func (c *component) GetRole(ctx context.Context, realmName string, roleID string
 	return roleRep, nil
 }
 
+func (c *component) CreateRole(ctx context.Context, realmName string, role api.RoleRepresentation) (string, error) {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	var roleRep = api.ConvertToKCRole(role)
+
+	locationURL, err := c.keycloakClient.CreateRole(accessToken, realmName, roleRep)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return "", err
+	}
+
+	//retrieve the role ID
+	reg := regexp.MustCompile(`[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}`)
+	roleID := string(reg.Find([]byte(locationURL)))
+
+	//store the API call into the DB
+	c.reportEvent(ctx, "API_ROLE_CREATION", database.CtEventRealmName, realmName, database.CtEventRoleID, roleID, database.CtEventRoleName, *role.Name)
+
+	return locationURL, nil
+}
+
+func (c *component) DeleteRole(ctx context.Context, realmName string, roleID string) error {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	role, err := c.keycloakClient.GetRole(accessToken, realmName, roleID)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return err
+	}
+
+	var roleName = *role.Name
+
+	err = c.keycloakClient.DeleteRole(accessToken, realmName, roleID)
```

Comment:
We must ensure we are deleting a business role

Reply:
Very good catch ! Corrected !